light README and train parameter changes

dribnet opened the pull request:
Two small and independent tweaks:
- The only small error I found getting things going was in the README - fuel-download doesn't need the size parameter, only fuel-convert does.
- Because training the CelebA classifier is so fast, I thought dropping the checkpointing and monitoring to every epoch made sense.

Also I'll use this space to note a few other things that might help others get started.

The packaged theanorc didn't work for me because I had some custom variables I needed (cuda root). But having the packaged template was good - it was easy to add my own variables to a copy and use that instead.

I also had to prepend all commands with `PYTHONPATH=.` to get imports to work.

I ran out of GPU memory training the VAE with discriminative regularization, but lowering the batch sizes was an effective workaround.

Since I was running this remotely, it wasn't clear why the `scripts` (like sample and reconstruct) were hanging indefinitely - then I checked and saw that it was using ipython. The code is already written to allow command line usage, the invocation is just different - use `python` explicitly and a `--save-path`:

```
THEANORC=theanorc.mine PYTHONPATH=. python scripts/sample --save-path sample_no_reg.png celeba_vae_no_regularization.zip
```

dribnet committed:
README: fixed fuel-download args

vdumoulin commented on experiments/train_celeba_classifier.py:
```diff
@@ -144,11 +144,11 @@ def run():
     valid_monitoring = DataStreamMonitoring(
         monitored_quantities, valid_monitor_stream, prefix="valid",
-        before_first_epoch=False, after_epoch=False, every_n_epochs=5)
+        before_first_epoch=False, after_epoch=False, every_n_epochs=1)
```
A simpler, equivalent setup would be `before_first_epoch=False, after_epoch=True`.

vdumoulin commented on experiments/train_celeba_classifier.py:
```diff
@@ -144,11 +144,11 @@ def run():
     # Prepare checkpoint
     checkpoint = Checkpoint(
-        'celeba_classifier.zip', every_n_epochs=5, use_cpickle=True)
+        'celeba_classifier.zip', every_n_epochs=1, use_cpickle=True)
```
Likewise: `'celeba_classifier.zip', after_epoch=True, use_cpickle=True`.

vdumoulin commented:
@dribnet Thanks! Some things to note:

You can override some of the settings in `theanorc` by using `THEANO_FLAGS=[...]`, e.g. on a multi-gpu machine, if you want to use `gpu1` explicitly, you could prefix your command with `THEANORC=theanorc THEANO_FLAGS=device=gpu1`.

I thought that the current working directory was always included in `sys.path`, but I can't confirm that with a quick Google search.

I ran the experiments on a Titan X and I didn't think of checking the memory usage. I think your tip regarding lowering batch size to decrease memory usage would have its place in the README if you don't mind adding it.

The scripts will display plots on screen by default, which means you need to enable X forwarding, otherwise you won't be able to see what's displayed. The reason why `ipython` is used instead of `python` is because I've had issues with `matplotlib` and `python` on OSX in the past, which seem to disappear if I switch to `ipython`. Once again, I think this would be worth mentioning in the README if you don't mind adding it.

dribnet commented:
@vdumoulin - thanks for looking this over. I'll clean this up soon and migrate some of my notes to the README as suggested.

dribnet committed:
README: added some notes on memory and ipython

Added notes on how to adjust batchsize to reduce memory
during training and how to use evaluation scripts without
ipython.

dribnet commented:
Branch updated with simple README notes as suggested.

I also removed the commit that adjusts the checkpointing and monitoring frequency during training as this is a separate idea and is probably unnecessary. We can put it back, or it might be clearer to just add this explicitly to the cli. I'll post a separate pull request with this and a few other training ideas.
